cogs/mod.py

```diff
@@ -654,6 +654,54 @@ async def approve(self, ctx, alias: str, invite: discord.Invite, times: int = 1)
         await self.bot.invitefilter.add(code=code, name=invite.guild.name, alias=alias, uses=times)
         await ctx.send(f"Approved an invite to {invite.guild}({code}) for posting {times} times")
         await self.bot.channels['mod-logs'].send(f"⭕ **Approved**: {ctx.author.mention} approved server {invite.guild}({code}) to be posted {times} times")
+                       
+                       
+                       
+    @commands.group()
+    async def role(self, ctx):
+
+#        self.role_list = {
+#            "cade": "cade enthusiast 😻",
+#            "streamer": "streamer(temp)",
+#        }
+        self.role_list = [
+            "cade enthusiast 😻",
+            "streamer(temp)",
+        ]
+
+        if ctx.invoked_subcommand is None:
+            await ctx.send_help(ctx.command)
+
+    @is_staff("HalfOP")
+    @role.command(name='give')
+    async def give_role(self, ctx, member: FetchMember, *, role: str):
+        if member is None:
+            return await ctx.send("Please include a user to add a role to.")
+
+        selected_role = [x for x in self.role_list if role in x]
+
+        if len(selected_role) > 1:      
```

Comment:
what if there was no match and selected_role is a empty list? then you are getting a index error when trying to add the role